Support operation `delete` on CloudObjectStoreContainer

miha-plesko opened the pull request:
With this commit we add "Configuration" menu option to the list of CloudObjectStore containers where we support operation `delete`.

This commit brings a new mid-page menu item "Configuration->Remove Object Storage Container" to the following pages:

* cloud object container details page
![capture2](https://cloud.githubusercontent.com/assets/8102426/23555597/d0e72fd8-0029-11e7-94b9-57afeabd8532.PNG)


* cloud object container full list
![capture](https://cloud.githubusercontent.com/assets/8102426/23555167/255ee9ae-0028-11e7-8e08-0ea9fee23b82.PNG)

* cloud object container list per StorageManager
![capture_1](https://cloud.githubusercontent.com/assets/8102426/23555561/afe71c8a-0029-11e7-8e67-fd6e01687678.PNG)



Video:
[delete_bucket.zip](https://github.com/ManageIQ/manageiq-ui-classic/files/783641/delete_bucket.zip)

Links:
* https://github.com/ManageIQ/manageiq/pull/13965 (merged)
* https://github.com/ManageIQ/manageiq-providers-amazon/pull/144 (merged)

martinpovolny commented:
I have not read the code yet, but I can see that there's a bunch of Rubocop issues and there are no tests for the newly added functions. So marking this as WIP for now.

miha-plesko commented:
Hello @martinpovolny I've added a dozen of unit tests and fixed rubocop issues. I'm not sure, however, whether it's better now that there is a mixed code styling with less rubocop issues or was it more readable before when code styling was consistent across the file (e.g. https://github.com/ManageIQ/manageiq-ui-classic/pull/420/files#diff-4b513b358c3d87df7b0be7e0eef4d95e).

Could you please take a look if there is anything else that can be improved? :D

Ladas commented on app/helpers/application_helper/toolbar/cloud_object_store_container_center.rb:
```diff
@@ -1,4 +1,27 @@
+            N_('Remove Object Storage Container'),
+            :url_parms => "main_div",
+            :confirm   => N_("Warning: The selected Object Storage Container and ALL related Objects will be "\
+                             "permanently removed!"),
```
we are missing supports check https://github.com/Ladas/manageiq-ui-classic/blob/29e35d051703fc0b1e85ec88d14ca62d972dcad5/app/helpers/application_helper/toolbar/cloud_subnet_center.rb#L25-L25

similar supports check should be also in the controller @romanblanco  ?

romanblanco replied:
@Ladas The support check is in the [Toolbar Button code](https://github.com/Ladas/manageiq-ui-classic/blob/29e35d051703fc0b1e85ec88d14ca62d972dcad5/app/helpers/application_helper/button/generic_feature_button_with_disable.rb#L11):


In Toolbar is only specified feature.

miha-plesko commented:
I've noticed an unused function in my commit so I've removed it and repushed.

martinpovolny commented:
@miha-plesko : the failed tests seem to be relevant and related to these changes. Please, take a look,

miha-plesko commented:
@martinpovolny I think all failed tests are a result of related PR in `manageiq` repo not being merged yet. Knowing this, I would still be happy to be hearing your opinion on the approach in this PR. This way we will be able to merge this one just after the related one is merged and tests turn green 😄

gberginc commented:
@miha-plesko close & reopen in case the merged PR (https://github.com/ManageIQ/manageiq/pull/13965) affects this.

miha-plesko commented:
@martinpovolny any suggestion why Travis fails here
https://travis-ci.org/ManageIQ/manageiq-ui-classic/jobs/207300784#L608
while it's all green locally (tested on my laptop and friend's laptop)?

The easiest workaround would be to delete the two tests that are failing, but I would feel better if we somehow make them green instead 😄

sergio-ocon commented:
Hi @martinpovolny The code is ready to merge, could you please have a look at it?

Thanks,
Sergio

martinpovolny commented on app/controllers/application_controller/ci_processing.rb:
```diff
@@ -1840,6 +1840,49 @@ def vm_button_operation(method, display_name, partial_after_single_selection = n
     vms.count
   end
 
+  def process_cloud_object_storage_buttons
```
can you, please, pass in `params[:pressed]` as an argument here?
(better for testing, code readability etc.)

martinpovolny commented:
This looks good! Please, just add the argument mentioned above.

The Rubocops about complexity we don't really enforce at this point, but I don't see the reason for the compaints in `spec/controllers/cloud_object_store_container_spec.rb`: 
```:exclamation: - Line 96, Col 7 - Rails/HttpPositionalArguments - Use keyword arguments instead of positional arguments for http call: post.```

I see you have hash arguments there, which looks good to me. So we'll ignore that too.

miha-plesko commented:
Thank you for the review @martinpovolny, I totally agree with your suggestions. I've applied them and repushed, please take a look.

martinpovolny commented:
Thx!
